src/field/pokemon.ts

```diff
@@ -4863,8 +4863,8 @@ export class PokemonBattleSummonData {
 export class PokemonTurnData {
   public flinched: boolean = false;
   public acted: boolean = false;
-  public hitCount: number;
-  public hitsLeft: number;
+  public hitCount: number = 0;
+  public hitsLeft: number = -1;
```

Comment:
Q:
Why `-1` here?
I mean `0` also means there is no more hits left, right?

Reply:
`0` makes the move assume it's finished and so multi-hit moves don't do all their hits, `-1` means it should calculate how many hits are left.

Reply:
@DayKev Would you mind adding that as a comment on top of this?
```suggestion
  /** 0 = Move is finished. -1 = flagged to calculate how many hits are left */
  public hitsLeft: number = -1;
```